.github/workflows/wheels.yml

```diff
@@ -62,16 +71,29 @@ jobs:
         run: |
           mv NOTICE_DEFAULT THIRD-PARTY-LICENSES
 
+      # actions/setup-python requires /Users/runner/hostedtoolcache to exist to work properly
+      # with macosx due to fixed shared library path
+      # https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#macos
+      - name: Create hostedtoolcache dir for macosx
+        if: ${{ matrix.builder.kind == 'macosx' }}
+        run: |
+          mkdir -p /Users/runner/hostedtoolcache
+
       # actions/setup-python doesn't yet support ARM
       # https://github.com/actions/setup-python/issues/678
-      - if: ${{ matrix.builder.arch != 'aarch64' }}
-        uses: actions/setup-python@v4
+      - name: Setup Python
+        if: ${{ matrix.builder.arch != 'aarch64' }}
+        uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
 
       - name: Install pipx
         run: |
-          python -m pip install --upgrade pipx 
+          which python
```

Comment:
Are these `which` lines still wanted?

Reply:
I used them for debugging, but left them on purpose, as they helped to understand the problem, and I am expecting changes for pip setup in the upcoming runner updates.